include/pnet_api.h

```diff
@@ -1172,23 +1172,51 @@ typedef struct pnet_ethaddr
 
 
 /**
- * LLDP configuration for a single port.
+ * Network interface
  */
-typedef struct pnet_lldp_port_cfg
+typedef struct pnet_netif
 {
+   char if_name[PNET_INTERFACE_NAME_MAX_SIZE]; /**< Terminated string */
+   pnet_ethaddr_t eth_addr;                    /**< Interface MAC address */
+} pnet_netif_t;
+
+/**
+ * Port configuration including network interface and
+ * lldp configuration parameters
+ */
+typedef struct pnet_port_cfg
+{
+   pnet_netif_t netif;
    char port_id[PNET_LLDP_PORT_ID_MAX_SIZE]; /**< Terminated string */
-   pnet_ethaddr_t port_addr;
    uint16_t rtclass_2_status;
    uint16_t rtclass_3_status;
-} pnet_lldp_port_cfg_t;
+} pnet_port_cfg_t;
 
 /**
- * LLDP configuration used by the Profinet stack.
+ * IP Configuration
  */
-typedef struct pnet_lldp_cfg
+typedef struct pnet_ip_cfg
 {
-   pnet_lldp_port_cfg_t ports[PNET_MAX_PORT];
-} pnet_lldp_cfg_t;
+   bool dhcp_enable; /**< Not supported by stack. */
+   pnet_cfg_ip_addr_t ip_addr;
+   pnet_cfg_ip_addr_t ip_mask;
+   pnet_cfg_ip_addr_t ip_gateway;
+} pnet_ip_cfg_t;
+
+/**
```

Comment:
Perhaps use Profinet vocabulary instead, where "interface" is an abstract group of ports?

Reply:
I tried some renaming and rewording. Hopefully to the better.